enhancements/windows-containers/windows-machine-config-operator-upgrades.md

```diff
@@ -0,0 +1,191 @@
+created for Windows machines. WMCO will not render more than the specified
+amount of Windows nodes unavailable.
+
+The procedure for an upgrade is as follows:
+1) A new WMCO version is released
+2) If the current cluster version fufills the minimum Kubernetes version
+   requirement, OLM upgrades WMCO. If the cluster version is not high enough,
+   the WMCO upgrade will occur once it is.
+3) The new WMCO reconciles as usual, ensuring that all unconfigured Windows
+   Machines are configured and join the cluster as a node. Each of them are
+   given an annotation indicating the WMCO version that configured them.
+4) Each Windows node is checked for the WMCO version annotation, if the
+   annotated version of a Windows node does not match the WMCO version, the
+   kubelet on the node is stopped.
+5) When the Windows Machine is destroyed and recreated by the Machine API
+   Operator, WMCO will reconcile again and configure the VM. This will
+   repeat until all Windows nodes have been configured by the upgraded WMCO.
+
+### User Stories
+
+Stories can be found within the [Windows Upgrades epic](https://issues.redhat.com/browse/WINC-404)
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces a change that
+  breaks the previous version of WMCO, Windows workloads will be interrupted
+  for a period of time before WMCO is upgraded and rolls out the new nodes.
+  This will be known ahead of release and can be messaged out appropriately.
+  For unexpected complications, we will add alerts indicating nodes failing
+  after upgrades.
+* OLM does not allow for easy downgrading. The downgrade process is recommended
+  to be done by removing the operator and reinstalling the previous version. If
+  a new version of an operator causes breaking issues for a user they will have
+  to perform this manual process.
+* If a MachineHealthCheck is not set up to watch Windows Machines, the upgrade
+  process will halt until the check is created. To mitigate this, an alert will
+  be generated to inform the user when a MachineHealthCheck for the Windows
+  MachineSet needs to be created. In order to make this health check
+  discoverable by WMCO, the MachineHealthCheck must specifically select the
+  Windows MachineSet through a label selector, matching the
+  "machine.openshift.io/cluster-api-machineset" label with the name of the
+  Windows MachineSet.
+
+### Test Plan
+
+* It is planned to use the upcoming CI For Optional Operators 
+  [Container Verification Pipeline](https://issues.redhat.com/browse/DPTP-900)
+  for release candidate testing and [PR-Based Testing Workflow](https://issues.redhat.com/browse/DPTP-1023)
+  for testing upgrades on a PR basis.
+* If the above epics are not completed in time, we will manually test WMCO
+  upgrades for each release.
+* Specifically we will test normal functionality after performing an upgrade
+  with the existing WMCO end-to-end test suite, as well as checking that
```

Comment:
I think this is too high level, can you be a bit more concrete on how you intend to do that? There will be a separate upgrade test suite which is outside of the WMCO operator which ensures that the upgrade is smooth?